types/index.d.ts

```diff
@@ -52,9 +52,12 @@ export interface BaseOptions {
   onLanguageSwitched?: (oldLocale: string, newLocale: string) => void
 }
 
+export type DisableDefaultRedirect = boolean | string[]
+
 export interface Options extends BaseOptions {
   baseUrl?: string | ((context: NuxtContext) => string)
   detectBrowserLanguage?: DetectBrowserLanguageOptions | false
+  disableDefaultRedirect?: DisableDefaultRedirect;
```

Comment:
The naming of that option is a bit too generic for my taste since it applies only in very specific case, only when using specific strategy. We can think more about it after other changes are addressed.

Reply:
What about 
```
type PrefixAndDefaultRule = 'default' | 'prefix'

interface PrefixAndDefaultRules {
  switchLocal: PrefixAndDefaultRule // 'default'
  routing: PrefixAndDefaultRule // 'default'
}

i18n: {
  prefixAndDefaultRules?: PrefixAndDefaultRules
}
```